Subject: [PATCH] Silence compiler warning and possible build break
From: Carsten Schlote <c.schlote@konzeptpark.de>

Changed code to compile without warnings. Broke my build with
-Werror set as default option by GCC (Ubuntu 8.10)

Signed-off-by: Carsten Schlote <c.schlote@konzeptpark.de>
---
 src/genpart.c |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

Index: genpart-1.0.1/src/genpart.c
===================================================================
--- genpart-1.0.1.orig/src/genpart.c
+++ genpart-1.0.1/src/genpart.c
@@ -40,6 +40,7 @@ int main(int argc, char *argv[])
 	struct partition_entry entry;
 	int opt, do_magic = 0;
 	unsigned short magic = 0xaa55;
+	int rc = 10;
 
 	struct option long_options[] = {
 		{ "help", no_argument, 0, 'h' },
@@ -87,10 +88,11 @@ int main(int argc, char *argv[])
 		}
 	}
 
-	fwrite(&entry, sizeof(struct partition_entry), 1, stdout);
-
-	if(do_magic)
-		fwrite(&magic, 2, 1, stdout);
-
-	return 0;
+	if (fwrite(&entry, sizeof(struct partition_entry), 1, stdout) > 0) {
+		if (do_magic) {
+			if (fwrite(&magic, 2, 1, stdout) > 0)
+				rc=0;
+		}
+	}
+	return rc;
 }
